test/new-e2e/tests/process/testing.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process contains end-to-end tests for the general functionality of the process agent.
+package process
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+)
+
+//go:embed config/process_check.yaml
+var processCheckConfigStr string
+
+//go:embed config/process_discovery_check.yaml
+var processDiscoveryCheckConfigStr string
+
+//go:embed config/system_probe.yaml
+var systemProbeConfigStr string
+
+// assertRunningChecks asserts that the given process agent checks are running on the given VM
+func assertRunningChecks(t *assert.CollectT, vm client.VM, checks []string, withSystemProbe bool) {
+	status := vm.Execute("sudo datadog-agent status --json")
+	var statusMap struct {
+		ProcessAgentStatus struct {
+			Expvars struct {
+				EnabledChecks                []string `json:"enabled_checks"`
+				SysProbeProcessModuleEnabled bool     `json:"system_probe_process_module_enabled"`
+			} `json:"expvars"`
+		} `json:"processAgentStatus"`
+	}
+	err := json.Unmarshal([]byte(status), &statusMap)
+	assert.NoError(t, err, "failed to unmarshal agent status")
+
+	assert.ElementsMatch(t, checks, statusMap.ProcessAgentStatus.Expvars.EnabledChecks)
+
+	if withSystemProbe {
+		assert.True(t, statusMap.ProcessAgentStatus.Expvars.SysProbeProcessModuleEnabled,
+			"system probe process module not enabled")
+	}
+}
+
+// assertStressProcessCollected asserts that the stress process is collected by the process check
+// and that it has the expected data populated
+func assertStressProcessCollected(
+	t *testing.T, payloads []*aggregator.ProcessPayload, withIOStats bool,
+) {
+	var found, populated bool
+	for _, payload := range payloads {
+		for _, process := range payload.Processes {
+			if len(process.Command.Args) > 0 && process.Command.Args[0] == "stress" {
+				found = true
+				populated = processHasData(process)
+
+				if withIOStats {
+					populated = populated && processHasIOStats(process)
+				}
+
+				if populated {
+					break
+				}
+			}
+		}
+
+		if found && populated {
+			break
+		}
+	}
+
+	require.True(t, found, "stress process not found")
+	assert.True(t, populated, "no stress process had all data populated")
```

Comment:
I guess same for the IO stats since that goes into the check.

Reply:
That's a good idea. I added logging functionality to dump the payloads if any assertion fails.

The dump is a bit messy (there's no straightforward way to pretty print structs), but I think it's better that the output is there if we need it rather than trying to make it pretty and susceptible to failure.